src/Motor.Extensions.Hosting.Kafka/KafkaMessageConsumer.cs

```diff
@@ -76,9 +76,13 @@ public async Task ExecuteAsync(CancellationToken token = default)
                 try
                 {
                     var msg = _consumer?.Consume(token);
-                    if (msg != null && !msg.IsPartitionEOF)
+                    if (msg is { IsPartitionEOF: false })
                     {
-                        SingleMessageHandling(token, msg);
+                        Task<bool> completed;
+                        do
+                        {
+                            completed = SingleMessageHandlingAsync(token, msg);
+                        } while (_options.SynchronousMessageHandling && !await completed);
```

Comment:
Are we sure that we want to instantly retry? We expect the Callback to use some sort of delay between its own retries (like Polly with HTTP requests), right? Otherwise I would expect some sort of delay with a policy here. Maybe even use Polly here?

Reply:
Polly is probably a good idea, since a reason for a temporary failure could be a HTTP 429 within the message handler which would never recover without a proper retry policy. Good idea!